<launch>

    <!-- 
        The topic that we'll send the commands to.
        By default, we want want to directly control the robot
     -->
    <arg name="output_topic" default="/robot/cmd_vel" />

    <!-- Launch GPS Decision -->
    <node name="gps_decision" pkg="decision" type="gps_decision" output="screen">
        <!-- The max linear (forward) speed -->
        <rosparam param="max_linear_speed">1.0</rosparam>

        <!-- The max angular (rotational) speed -->
        <rosparam param="max_angular_speed">0.7</rosparam>

        <!-- How much effect the distance to the waypoint has on the linear speed -->
        <rosparam param="linear_distance_factor">0.1</rosparam>

        <!-- How much effect the angle between our heading and the -->
        <!-- heading we should be on to get the the waypoint has on our linear speed -->
        <rosparam param="linear_heading_factor">1.0</rosparam>

        <!-- How much effect the distance to the waypoint has on the angular speed -->
        <rosparam param="angular_distance_factor">0.2</rosparam>

        <!-- How much effect the heading between our heading and the -->
        <!-- heading we should be on to get the the waypoint has on our angular speed -->
        <rosparam param="angular_heading_factor">0.4</rosparam>

        <remap from="/gps_decision/twist" to="$(arg output_topic)"/>
    </node>

</launch>
